llvm/lib/Transforms/Utils/SplitModule.cpp

```diff
@@ -275,8 +278,8 @@ void llvm::SplitModule(
     ValueToValueMapTy VMap;
     std::unique_ptr<Module> MPart(
         CloneModule(M, VMap, [&](const GlobalValue *GV) {
-          if (ClusterIDMap.count(GV))
-            return (ClusterIDMap[GV] == I);
+          if (auto It = ClusterIDMap.find(GV); It != ClusterIDMap.end())
```

Comment:
For readability it would probably be better to move out the assignment of It.

Reply:
Would it? It's a common pattern that reduces the scope of 'It': https://github.com/llvm/llvm-project/blob/9d45077df92a64ff8db5b249ac2f2d45e82ab73c/clang/lib/AST/Interp/Program.cpp#L117, https://github.com/llvm/llvm-project/blob/9d45077df92a64ff8db5b249ac2f2d45e82ab73c/llvm/lib/IR/DebugLoc.cpp#L79, https://github.com/llvm/llvm-project/blob/9d45077df92a64ff8db5b249ac2f2d45e82ab73c/clang/lib/Analysis/FlowSensitive/Formula.cpp#L60 etc.

Reply:
I also use this pattern a lot and I think it's more readable than 2 separate line

Reply:
Right I guess sits down to personal preference.